### DISCUSSION TITLE
Feature Request: Tooltips on Hover Dates/Bars on Textual-Plotext

### DISCUSSION THREAD

**Discussion by ADevAJ** [Textualize/textual-plotext#9] [Ideas]
Really enjoying using textual-plotext! It opens up a whole new dimension for textual applications. 

Would love to see textual's Tooltips functionality extended to textual-plotext.

Where mouse hovering over a specific bar / datetime / candlestick / within the PlotextPlot
can  triggers a callback to format a tooltip  at the cursor location. 

== Textual as a Plotly / Dash replacement ?

**Comment by davep** [comment]
It's a nice idea but we don't have any plans like this for this library, at least not at the moment. We do have a rich set of plots that we'd likely add such functionality to [on our roadmap](https://textual.textualize.io/roadmap/), but I think we envisage that more as something that we'd build from the ground up. This library is more aimed at making it easy and straightforward to embed Plotext plots in a Textual application.

**Reply by ADevAJ** [reply]
How much do you think would be involved in adding this feature to Textual-Plotext ? 

Would you be open to the possibility of accepting a PR for this purpose if I am willing take on the workload? I have some time over the holidays and think this could be a good project for me.

**Reply by davep** [reply]
My concern would be that, even if done, it would potentially be problematic to maintain, given it would likely need a lot of downstream hacking into the upstream internals. 

I sense this is work better left for a Textual-first plotting library. Unless, that is, you can see a method of doing this that wouldn't be overly-vulnerable to upstream changes?
